mibidata/mibi_image.py

```diff
@@ -185,6 +185,54 @@ def __init__(self, data, channels, **kwargs):
             setattr(self, k, v)
             self._user_defined_attributes.append(k)
 
+    def add_attr(self, **kwargs):
+        """Adds metadata key-value pairs as attributes to
+           the class instance in use. If attribute already exists
+           for the current instance, updates value of attribute.
+
+        Args:
+            kwargs: A mapping of arguments for a user to add multiple
+                    attributes with their respecitve
+                    values.
+        """
+        date = kwargs.pop('date', self.date)
+        datetime_format = kwargs.pop('datetime_format', _DATETIME_FORMAT)
+        try:
+            self.date = datetime.datetime.strptime(date, datetime_format)
+        except TypeError:  # Given as datetime obj already, or None.
+            self.date = date
+        for key, value in kwargs.items():
+            if key not in SPECIFIED_METADATA_ATTRIBUTES:
```

Comment:
My opinion is that in an "add attribute" function, if the attribute already exists, an error should be thrown. That, to me, seems more intuitive. I would, before setting anything, check the keys in `kwargs` and if they are already in the metadata, raise an error.